vault/plugin_catalog.go

```diff
@@ -41,78 +42,113 @@ type PluginCatalog struct {
 	logger          log.Logger
 
 	// multiplexedClients holds plugin process connections by plugin name
-	// This allows a single grpc connection to communicate with multiple
-	// databases. Each database configuration using the same plugin will be
-	// routed to the existing plugin process.
+	//
+	// This allows plugins that suppport multiplexing to use a single grpc
+	// connection to communicate with multiple "backends". Each backend
+	// configuration using the same plugin will be routed to the existing
+	// plugin process.
 	multiplexedClients map[string]*MultiplexedClient
 
 	lock sync.RWMutex
 }
 
+// PluginClient represents a connection to a plugin process
 type PluginClient struct {
-	logger   log.Logger
-	id       string
-	protocol plugin.ClientProtocol
+	logger log.Logger
+
+	// id is the connection ID
+	id string
+
 	// client handles the lifecycle of a plugin process
-	client *plugin.Client
+	// multiplexed plugins share the same client
+	client      *plugin.Client
+	clientConn  *pluginClientConn
```

Comment:
Could this be `grpc.ClientConnInterface`?

Reply:
Yes, that works.

Reply:
yes, that works and should make it more flexible.

Reply:
addressed in: https://github.com/hashicorp/vault/pull/13896/commits/3afa08873ad064657cadc39d9bce94e45f49392e